src/app/store/machine/slice.ts

```diff
@@ -1015,33 +1016,104 @@ const machineSlice = createSlice({
+      }),
+      reducer: (
+        state: MachineState,
+        action: PayloadAction<
+          MachineStateDetailsItem["errors"],
+          string,
+          GenericItemMeta<{ system_id: Machine[MachineMeta.PK] }>
+        >
+      ) => {
+        if (action.meta.requestId && action.meta.requestId in state.details) {
+          state.details[action.meta.requestId].errors = action.payload;
+          state.details[action.meta.requestId].loading = false;
+        }
+        state = setErrors(state, action, "get");
+      },
     },
-    getStart: (state: MachineState) => {
-      state.loading = true;
+    getStart: {
+      prepare: (
+        item: { system_id: Machine[MachineMeta.PK] },
+        requestId: string
+      ) => ({
+        meta: {
+          item,
+          requestId,
+        },
+        payload: null,
+      }),
+      reducer: (
+        state: MachineState,
+        action: PayloadAction<
+          MachineStateDetailsItem["errors"],
+          string,
+          GenericItemMeta<{ system_id: Machine[MachineMeta.PK] }>
+        >
+      ) => {
+        if (action.meta.requestId) {
+          if (action.meta.requestId in state.details) {
+            state.details[action.meta.requestId].loading = true;
+          } else {
+            state.details[action.meta.requestId] = {
+              errors: null,
+              loaded: false,
+              loading: true,
+              system_id: action.meta.item.system_id,
+            };
+          }
+        }
+      },
     },
-    getSuccess: (state: MachineState, action: PayloadAction<Machine>) => {
-      const machine = action.payload;
-      // If the item already exists, update it, otherwise
-      // add it to the store.
-      const i = state.items.findIndex(
-        (draftItem: Machine) => draftItem.system_id === machine.system_id
-      );
-      if (i !== -1) {
-        state.items[i] = machine;
-      } else {
-        state.items.push(machine);
-        // Set up the statuses for this machine.
-        state.statuses[machine.system_id] = DEFAULT_STATUSES;
-      }
-      state.loading = false;
+    getSuccess: {
+      prepare: (
+        item: { system_id: Machine[MachineMeta.PK] },
+        requestId: string,
+        machine: MachineDetails
+      ) => ({
+        meta: {
+          item,
+          requestId,
+        },
+        payload: machine,
+      }),
+      reducer: (
+        state: MachineState,
+        action: PayloadAction<
+          MachineDetails,
+          string,
+          GenericItemMeta<{ system_id: Machine[MachineMeta.PK] }>
+        >
+      ) => {
+        const machine = action.payload;
+        // If the item already exists, update it, otherwise
+        // add it to the store.
+        const i = state.items.findIndex(
+          (draftItem: Machine) => draftItem.system_id === machine.system_id
+        );
+        if (i !== -1) {
+          state.items[i] = machine;
+        } else {
+          state.items.push(machine);
```

Comment:
Could you perform this operation this without mutating the original array/object. https://redux.js.org/faq/immutable-data

Reply:
Redux Toolkit uses immer behind the scenes so while this looks like a mutation it's actually using using Immer's `produce` when executing the change: https://redux-toolkit.js.org/usage/immer-reducers

Reply:
Oh interesting, I had no idea redux toolkit already uses Immer behind the scenes.